### DISCUSSION TITLE
Define workflow for managing dependencies

### DISCUSSION THREAD

**Discussion by slint** [plazi/lycophron#6] [Tech]
There are a couple of solutions already out there:

- [x] [`pip-tools`](https://github.com/jazzband/pip-tools) + virtualenv (I personally have a preference for this one, although there is some "manual" work involved to maintain, but can be documented).
- [ ] [`pipenv`](https://pipenv.pypa.io/en/latest/)
- [ ] [Poetry](https://python-poetry.org/)
- [ ] [PDM](https://pdm.fming.dev/)

**Comment by mguidoti** [comment]
If you're more comfortable with the first option, let's go with it. I'm used to use `pipenv`, but I'm always open to learn.

We had a brief discussion about this topic before in https://github.com/plazi/lycophron/issues/2.

**Reply by mguidoti** [reply]
In the end, I think we are going with a mixed between `pip-toosl` and `poetry` since the `pyproject.toml` is present in the initial PR by @ctrl-alt-soneca, right?

I'm ok with that, and already reproduced the environments locally.
